tests/functions_tests/test_tensor_network.py

```diff
@@ -0,0 +1,175 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+if cuda.available:
+    cuda.init()
+
+
+class TestTensorNetwork(unittest.TestCase):
+
+    def setUp(self):
+        in_shape = (2, 3)
+        out_size = 4
+
+        self.f = functions.TensorNetwork(in_shape, out_size)
+        self.f.W = numpy.random.uniform(
+            -1, 1, self.f.W.shape).astype(numpy.float32)
+        self.f.V1 = numpy.random.uniform(
+            -1, 1, self.f.V1.shape).astype(numpy.float32)
+        self.f.V2 = numpy.random.uniform(
+            -1, 1, self.f.V2.shape).astype(numpy.float32)
+        self.f.b = numpy.random.uniform(
+            -1, 1, self.f.b.shape).astype(numpy.float32)
+        self.f.zero_grads()
+
+        self.W = self.f.W.copy()
+        self.V1 = self.f.V1.copy()
+        self.V2 = self.f.V2.copy()
+        self.b = self.f.b.copy()
+
+        batch_size = 10
+        self.e1 = numpy.random.uniform(
+            -1, 1, (batch_size, in_shape[0])).astype(numpy.float32)
+        self.e2 = numpy.random.uniform(
+            -1, 1, (batch_size, in_shape[1])).astype(numpy.float32)
+        self.gy = numpy.random.uniform(
+            -1, 1, (batch_size, out_size)).astype(numpy.float32)
+
+        self.y = (
+            numpy.einsum('ij,ik,jkl->il', self.e1, self.e2, self.W) +
+            self.e1.dot(self.V1) +
+            self.e2.dot(self.V2) +
+            self.b)
+
+    def check_forward(self, x_data):
```

Comment:
It is just because functions of the same name in other functions have such arguments. I agree with your proposal because we can make use of the information that the number of input tensors is 2.